vllm_spyre/v1/core/scheduler.py

```diff
@@ -224,11 +233,29 @@ def can_schedule(self, request) -> bool:
         max_prompt_batch_size = 1
         max_context_len = self.scheduler_config.max_model_len
 
+        # if the batch is locked by a request which has been waiting for
+        # longer then VLLM_SPYRE_MAX_WAITING_TIME_SECONDS, we cannot
+        # schedule the current sequence until we have served this request
+        if self.batch_is_locked:
+            return False
```

Comment:
instead of locking the batch entirely, shouldn't we just disallow any skipping of requests in the queue until the request at the head of the waiting queue schedules?

I haven't followed super closely but my assumption is that the blocked request may be able to be scheduled before the full batch finishes. E.g. with the 128k limit, a 64k request could potentially schedule once the batch has drained down to a single other request, so we wouldn't need to wait for the last one to finish.